src/ProcessAbstract.php

```diff
@@ -105,7 +105,12 @@ public function exit(): void
     public function shutdown(): ProcessInterface
     {
         if ($this->_read(self::PROP_IS_SHUTDOWN_METHOD_ACTIVE)) {
-            $this->_getLogger()->critical("Shutdown method invoked.");
+            // to avoid hitting artificial memory limits while executing this block
+            ini_set('memory_limit','-1');
+            $this->_getLogger()->critical(
+                'Shutdown method invoked.',
+                ['error_get_last' => error_get_last()]
```

Comment:
I like it. Is there any way that we can *further* bake the knowledge in that this may not be the trigger for invoking the shutdown of the Process? That is that there is no guarantee that this error was the cause? The answer may just be "no" but I can see how this may cause someone to incorrectly attribute cause and effect at some point. Pro's and cons.

Reply:
I couldn't find any way to determine whether an error caused the shutdown. With userspace being able to set its own `error_reporting(), it could very well be "this error was uncaught but didn't cause the shutdown". If you know of some other way, I'm very interested

Reply:
yeah, idk how else to bake in that message about cause and effect so that someone just looking at the logs understands that the `error_get_last` can be unrelated. I just *feel* like someone will eventually be looking for [DenverCoder9](https://xkcd.com/979/) and cursing us.